pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,143 @@ $title: Use custom JavaScript in AMP pages
+
+**1) Know what's supported.** Get to know [the rich set of supported APIs](https://github.com/ampproject/worker-dom/blob/main/web_compat_table.md). You need to think about JavaScript differently - more as a concise set of tools.
+
+**2) Use Preact.** Since the team couldn't rebuild all the DOM APIs, they focused on the limited set of APIs used by [Preact](https://preactjs.com/) and [React](https://reactjs.org/). If you use one of these frameworks, you should be able to build elaborate interactions with much less worry about what's supported. That said, React bundles often exceed `amp-script`'s [150K limit](../../../documentation/components/reference/amp-script.md#size-of-javascript-code). For this reason, Preact is strongly recommended.
+
+`amp-script` has been tested with frameworks like [Vue](https://vuejs.org/), [Angular](https://angularjs.org/), [Aurelia](https://aurelia.io/), and [lit-html](https://lit-html.polymer-project.org/), but less extensively. If you find a gap, please [file an issue](https://github.com/ampproject/worker-dom/issues) - or, better still, [submit a pull request](https://github.com/ampproject/worker-dom/pulls).
+
+Since `amp-script` cannot exhaustively support the DOM API, simply copying a library like [jQuery](https://jquery.com/) into an `<amp-script>` component will not work.
+
+## How should I use amp-script?
+
+`amp-script` was created to help you fill the gaps in your AMP webpage's functionality. One can't simply copy existing JavaScript into `amp-script` and expect it to work without modification. Instead, `amp-script` takes care of logic and interactions that existing AMP components don't provide.
 
-For user experience and security reasons, `amp-script` enforces DOM manipulation restrictions.
+The following are a few excellent use cases for `amp-script`.
 
-### User interaction
+### Create new interactions
 
-When a user interacts with elements wrapped inside an `<amp-script>` component, your custom JavaScript must return DOM manipulations quickly when needed. By default, changes to the DOM are permitted **less than one second** from the initial interaction. A notable exception is when your code must retrieve data from the network via `fetch`. Here DOM changes can be requested after the response is returned to the user and for **less than one second** afterward. If a script mutates the DOM outside of a permitted window, this will result in a fatal error and the `<amp-script>` component will terminate the Web Worker. A terminated `<amp-script>` component will not run again.
+When possible, it's better to use an AMP component for a DOM interaction. If the interaction you wish to create would be of interest to other developers, you are encouraged to [suggest and contribute](https://github.com/ampproject/amphtml/blob/master/CONTRIBUTING.md) a new component. Otherwise, `amp-script` provides a fine way to create interactions not currently supported by AMP components.
```

Comment:
This section starts by saying when not to use amp-script and uses passive voice. Suggested rewrite:
amp-script lets you create interactions that are not supported in AMP components. When possible, it's better to use an AMP component for a DOM interaction. If the interaction you wish to create would be of interest to other developers, we encourage you to suggest and contribute a new component.

Reply:
I was concerned about this section too. Thanks! I'll see if I can beef up the positive side with any new info too.